runelite-client/src/main/java/net/runelite/client/plugins/npchighlight/NpcIndicatorsPlugin.java

```diff
@@ -257,14 +257,25 @@ public void onMenuEntryAdded(MenuEntryAdded event)
 		{
 			type -= MENU_ACTION_DEPRIORITIZE_OFFSET;
 		}
-
-		if (config.highlightMenuNames() &&
-			NPC_MENU_ACTIONS.contains(MenuAction.of(type)) &&
-			highlightedNpcs.stream().anyMatch(npc -> npc.getIndex() == event.getIdentifier() && (!npc.isDead() || config.highlightDeadNpcs())))
+		if (NPC_MENU_ACTIONS.contains(MenuAction.of(type)))
 		{
+			String target;
+			final NPC menuNpc = client.getCachedNPCs()[event.getIdentifier()];
+			if (menuNpc.isDead() && config.recolorDeadNpcs())
```

Comment:
This logic should just be for choosing an color and not for building the menu target